File: root/tags/1.5.0/source/strategyagent/src/main/sample_data/samples/scripts/hello_world.rb
#
# author:anshul@marketcetera.com
# since 1.0.0
# version: $Id$
#
#
include_class "org.marketcetera.strategy.ruby.Strategy"

###############################
# Hello World Strategy        #
###############################
class HelloWorld < Strategy

    ##########################################
    # Executed when the strategy is started. #
    #                                        #
    # Use this method to set up data flows   #
    #  and other initialization tasks.       #
    ##########################################
    def on_start
      warn "Hello World!"
    end

    ############################################
    # Executed when the strategy is stopped.   #
    ############################################
    def on_stop
      warn "Good Bye!"
    end
end

File: root/trunk/source/strategyagent/src/main/sample_data/samples/java/scripts/HelloWorld.java
package sample;

import org.marketcetera.strategy.java.Strategy;


/* $License$ */
/**
 * Sample Hello World strategy.
 *
 * @author anshul@marketcetera.com
 * @version $Id$
 * @since 2.0.0
 */
public class HelloWorld extends Strategy {
    /**
     * Executed when the strategy is started.
     * Use this method to set up data flows
     * and other initialization tasks.
     */
    @Override
    public void onStart() {
        warn("Hello World!");
    }

    /**
     * Executed when the strategy is stopped.
     */
    @Override
    public void onStop() {
        warn("Good Bye!");
    }
}
